### DISCUSSION TITLE
Change glyph width in bitmap font for individual pixel sizes

### DISCUSSION THREAD

**Discussion by colorcube** [fontforge/fontforge#4650] [Q&A]
Hi

I have an OTF vector font. For several pixel sizes (12, 14, 15, ...) I created bitmap font. This is again saved as an OTF font.

It works all fine, but in some pixel sizes it would be needed to change the width of the glyph. For example "i" has too much space around in 14px.

I don't see how this can be done. In the pixel glyph edit window there are guide lines which can be moved but that don't change the glyph width.
In the menu of the pixel glyph edit window there's a menu entry "Metrics>Set Width...." but that's greyed out. 

UPDATE
As I understand when you edit a single bitmap font you can change the glyph width, but not when you combine bitmap and outline in a font.
I'm wondering if it's a limitation or FontForge or OTF/TTF (for example).

Any ideas?
Thanks

**Comment by skef** [comment]
>  In the pixel glyph edit window there are guide lines which can be moved but that don't change the glyph width

When I move the right guideline relative to the pixels and open a MetricsView window for  that strike size I see the change. How are you verifying that that doesn't work?

**Reply by colorcube** [reply]
Thanks to try it out!

I just found this:

_Set Width
This is only available if there is no outline font (ie. if you opened a bdf file directly)._

https://fontforge.org/docs/ui/mainviews/bitmapview.html#additional-tools-from-the-popup-menu

When I try that with a single bitmap font it works indeed. 
But combined with outlines changing the width has no effect.

Is this a limitation of font file formats or FontForge? I'm pretty shure the original Arial have bitmap fonts with different width in it.

Separating the bitmap fonts would be a bit inconvinent. Still I hope there may be a way...

**Reply by skef** [reply]
Are you sure that whatever program you're testing the font in is even using your bitmap strikes? Just because you include them doesn't force a program to use them, especially if it has the option of anti-aliasing instead.

For example, try drastically changing a bitmap and verifying that you see the changes.

**Reply by colorcube** [reply]
Yes it works very well. (Linux, freetype, ...). I can draw a smiley in one glyph of pixel size 14 and see exactly the smiley using with my application when size 14 is selected

**Reply by skef** [reply]
Then I'm afraid I'm out of my depth. One would have to trace the data in the font back through the output routines to determine why the width isn't changing. 

I did just verify that changing the width line is reflected in the BDFChar saved in the file, for whatever that's worth.

**Reply by colorcube** [reply]
Thanks for your help.
As I understand when you use a single bitmap font it works, but not when you combine bitmap and outline in a font.
I'm wondering if it's a limitation or FontForge or OTF/TTF (for example).
